src/user/index.ts

```diff
@@ -217,13 +217,59 @@ export const getUser = async (uuid: string) => {
 			...(await Oauth.getAuthHeaders())
 		}
 	})
-	if(result.status !== 200) {
-		throw new AuthError("Unable to get user");
+	if(result.status === 403) {
+		throw new ApiGetError("You do not have access to view this user. (403 Forbidden)");
+	}
+	else if(result.status !== 200) {
+		throw new ApiGetError("We were unable to get information about this user.");
 	}
 
 	return await result.json() as FullUser;
 }
 
+export const modifyUser = async (
+	uuid: string,
+	firstname?: string, 
+	lastname?: string, 
+	username?: string, 
+	email?: string, 
+	phone?: string,
+	guardian_phone?: string,
+	address?: string,
+	postal_code?: string,
+	birthdate?: string,
+	gender?: string,
+) => {
+const response = await fetch(`${getApiServer()}/user/${uuid}`, {
+	method: 'PATCH',
+	headers: {
+		'Content-Type': 'application/json',
+		...(await Oauth.getAuthHeaders())
+	},
+	body: JSON.stringify({
+		uuid,
+		firstname, 
+		lastname, 
+		username, 
+		email, 
+		phone,
+		guardian_phone,
+		address,
+		postal_code,
+		birthdate,
+		gender
+	})})
+	if (response.status === 403) {
+		throw new ApiPatchError("You do not have access to edit user information. (403 Forbidden)")
+	} 
+	else if (response.status !== 200) {
+		throw new ApiPatchError((await response.json())['error']);
```

Comment:
We need better error handling than this. But better to fix local, let's just merge this PR for now